model/riscv_sys_regs.sail

```diff
@@ -787,11 +785,31 @@ bitfield Envcfg : bits(64) = {
   FIOM   : 0,
 }
 
-register menvcfg : Envcfg
-register senvcfg : Envcfg
+bitfield SEnvcfg : xlenbits = {
+  // Cache Block Zero instruction Enable
+  CBZE   : 7,
+  // Cache Block Clean and Flush instruction Enable
+  CBCFE  : 6,
+  // Cache Block Invalidate instruction Enable
+  CBIE   : 5 .. 4,
+  // Reserved WPRI bits.
+  wpri_0 : 3 .. 1,
+  // Fence of I/O implies Memory
+  FIOM   : 0,
+}
+
+register menvcfg : MEnvcfg
+register senvcfg : SEnvcfg
+
+function legalize_menvcfg(o : MEnvcfg, v : bits(64)) -> MEnvcfg = {
+  let v = Mk_MEnvcfg(v);
+  let o = [o with FIOM = if sys_enable_writable_fiom() then v[FIOM] else 0b0];
+  // Other extensions are not implemented yet so all other fields are read only zero.
+  o
+}
 
-function legalize_envcfg(o : Envcfg, v : bits(64)) -> Envcfg = {
-  let v = Mk_Envcfg(v);
+function legalize_senvcfg(o : SEnvcfg, v : bits(64)) -> SEnvcfg = {
+  let v = Mk_SEnvcfg(v);
```

Comment:
Does this actually compile on RV32? You have `v` as 64-bit and `SEnvcfg` as 32-bit.

Reply:
Ah. v should also be xlenbits. I will fix that.

Reply:
This is fixed.